File: python/cliente.py
import socket

def cliente(host='127.0.0.1', port=12345):
    # Crear el socket (IPv4, TCP)
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as cliente_socket:
        # Conectar al servidor
        cliente_socket.connect((host, port))

        # Recibir el mensaje del servidor
        mensaje_servidor = cliente_socket.recv(1024)
        print(f"Mensaje del servidor: {mensaje_servidor.decode('utf-8')}")

        # Enviar un mensaje al servidor
        mensaje_cliente = "¡Hola servidor!"
        cliente_socket.sendall(mensaje_cliente.encode('utf-8'))

if __name__ == "__main__":
    cliente()

File: java/cliente.java
import java.io.*;
import java.net.*;

public class cliente {
    public static void main(String[] args) {
        String host = "127.0.0.1"; // localhost
        int puerto = 12345;

        try (Socket socket = new Socket(host, puerto)) {
            System.out.println("Conectado al servidor en " + host + ":" + puerto);

            // Flujo de entrada: leer mensaje del servidor
            BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
            String mensajeServidor = in.readLine();
            System.out.println("Mensaje del servidor: " + mensajeServidor);

            // Flujo de salida: enviar mensaje al servidor
            PrintWriter out = new PrintWriter(socket.getOutputStream(), true);
            out.println("¡Hola servidor!");

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
